src/appkit.cs

```diff
@@ -27427,4 +27575,265 @@ interface NSStepperTouchBarItem
+		bool ForPrinting { [Bind ("isForPrinting")] get; set; }
+
+		[Export ("createsNewApplicationInstance")]
+		bool CreatesNewApplicationInstance { get; set; }
+
+		[Export ("allowsRunningApplicationSubstitution")]
+		bool AllowsRunningApplicationSubstitution { get; set; }
+
+		[Export ("arguments", ArgumentSemantic.Copy)]
+		string[] Arguments { get; set; }
+
+		[Export ("environment", ArgumentSemantic.Copy)]
+		NSDictionary<NSString, NSString> Environment { get; set; }
+
+		[NullAllowed, Export ("appleEvent", ArgumentSemantic.Strong)]
+		NSAppleEventDescriptor AppleEvent { get; set; }
+
+		[Export ("architecture")]
+		int Architecture { get; set; }
+
+		[Export ("requiresUniversalLinks")]
+		bool RequiresUniversalLinks { get; set; }
+	}
+
+	[Mac (10,15)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface NSTextCheckingController
+	{
+		[Export ("initWithClient:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (INSTextCheckingClient client);
+
+		[Export ("client")]
+		INSTextCheckingClient Client { get; }
+
+		[Export ("invalidate")]
+		void Invalidate ();
+
+		[Export ("didChangeTextInRange:")]
+		void DidChangeText (NSRange range);
+
+		[Export ("insertedTextInRange:")]
+		void InsertedText (NSRange range);
+
+		[Export ("didChangeSelectedRange")]
+		void DidChangeSelectedRange ();
+
+		[Export ("considerTextCheckingForRange:")]
+		void ConsiderTextChecking (NSRange range);
+
+		[Export ("checkTextInRange:types:options:")]
+		void CheckText (NSRange range, NSTextCheckingTypes checkingTypes, NSDictionary<NSString, NSObject> options);
+
+		[Export ("checkTextInSelection:")]
+		void CheckTextInSelection ([NullAllowed] NSObject sender);
+
+		[Export ("checkTextInDocument:")]
+		void CheckTextInDocument ([NullAllowed] NSObject sender);
+
+		[Export ("orderFrontSubstitutionsPanel:")]
+		void OrderFrontSubstitutionsPanel ([NullAllowed] NSObject sender);
+
+		[Export ("checkSpelling:")]
+		void CheckSpelling ([NullAllowed] NSObject sender);
+
+		[Export ("showGuessPanel:")]
+		void ShowGuessPanel ([NullAllowed] NSObject sender);
+
+		[Export ("changeSpelling:")]
+		void ChangeSpelling ([NullAllowed] NSObject sender);
+
+		[Export ("ignoreSpelling:")]
+		void IgnoreSpelling ([NullAllowed] NSObject sender);
+
+		[Export ("updateCandidates")]
+		void UpdateCandidates ();
+
+		[Export ("validAnnotations")]
+		string[] ValidAnnotations { get; }
+
+		[Export ("menuAtIndex:clickedOnSelection:effectiveRange:")]
+		[return: NullAllowed]
+		NSMenu MenuAtIndex (nuint location, bool clickedOnSelection, NSRange effectiveRange);
+
+		[Export ("spellCheckerDocumentTag")]
+		nint SpellCheckerDocumentTag { get; set; }
+	}
+
+	delegate NSCollectionViewItem NSCollectionViewDiffableDataSourceItemProvider (NSCollectionView collectionView, NSIndexPath indexPath, NSObject itemIdentifierType);
+
+	delegate NSView NSCollectionViewDiffableDataSourceSupplementaryViewProvider (NSCollectionView collectionView, string str, NSIndexPath indexPath);
```

Comment:
Don't use acronyms for the delegates, is there a nicer parameter name so that it makes more sense for the users, 'str' is not very precise.